src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD011UseAsyncLazyAnalyzerTests.cs

```diff
@@ -0,0 +1,172 @@
+
+        [Fact]
+        public void ReportErrorOnLazyOfTConstructionInFieldValueTypeArg()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+
+class Test {
+    Lazy<Task<int>> t = new Lazy<Task<int>>();
+    Lazy<Task<int>> t2;
+    Lazy<int> tInt = new Lazy<int>();
+}
+";
+            this.expect.Locations = new[] {
+                new DiagnosticResultLocation("Test0.cs", 6, 25),
+            };
+            this.VerifyCSharpDiagnostic(test, this.expect);
+        }
+
+        [Fact]
+        public void ReportErrorOnLazyOfTConstructionInFieldRefTypeArg()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+
+class Test {
+    Lazy<Task<object>> t3 = new Lazy<Task<object>>();
+}
+";
+            this.expect.Locations = new[] {
+                new DiagnosticResultLocation("Test0.cs", 6, 29),
+            };
+            this.VerifyCSharpDiagnostic(test, this.expect);
+        }
+
+        [Fact]
+        public void JTFRunInLazyValueFactory_Delegate()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+class Test {
+    JoinableTaskFactory jtf;
+
+    void Foo() {
+        var t4 = new Lazy<int>(delegate {
+            jtf.Run(async delegate {
+                await Task.Yield();
+            });
+
+            return 3;
+        });
+    }
+}
+";
+            var expect = this.CreateDiagnostic(11, 13, 7);
+            this.VerifyCSharpDiagnostic(test, expect);
+        }
+
+        [Fact]
+        public void JTFRunInLazyValueFactory_Lambda()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+class Test {
+    JoinableTaskFactory jtf;
+
+    void Foo() {
+        var t4 = new Lazy<int>(() => {
+            jtf.Run(async delegate {
+                await Task.Yield();
+            });
+
+            return 3;
+        });
+    }
+}
+";
+            var expect = this.CreateDiagnostic(11, 13, 7);
+            this.VerifyCSharpDiagnostic(test, expect);
+        }
+
+        [Fact]
+        public void JTFRunInLazyValueFactory_MethodGroup()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+class Test {
+    JoinableTaskFactory jtf;
+
+    void Foo() {
+        var t4 = new Lazy<int>(LazyValueFactory);
+    }
+
+    int LazyValueFactory() {
+        jtf.Run(async delegate {
+            await Task.Yield();
+        });
+
+        return 3;
+    }
+}
+";
+
+            // We can change this to verify a diagnostic is reported if we ever implement this.
+            this.VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void ReportErrorOnLazyOfTConstructionInLocalVariable()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+
+class Test {
+    void Foo() {
+        var t4 = new Lazy<Task<object>>();
```

Comment:
Is there a need to test:

new Lazy<Task<int>>(async () => await JTF.RunAsync(...));

I am guessing that's covered already due to return type but wanted to check.